(commit amended to add packagekitqt6.pc.in)

From 510629798c77c5723339876d4a43aa8f3627581a Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Fri, 2 Jul 2021 15:39:49 +0200
Subject: [PATCH] Add build system support for Qt6

Allow building against Qt6 by replacing the hardcoded major version number with a variable

No code changes are needed for Qt6 support

---
 CMakeLists.txt                            |  4 +++-
 src/CMakeLists.txt                        | 25 ++++++++++++++---------
 src/modules/CMakeLists.txt                | 12 +++++------
 src/modules/packagekit-qt-config.cmake.in |  4 ++--
 src/packagekitqt6.pc.in                   | 11 ++++++++++
 5 files changed, 37 insertions(+), 19 deletions(-)
 create mode 100644 src/packagekitqt6.pc.in

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 08787de..a1afc40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,8 @@ include(GNUInstallDirs)
 
 set(QPACKAGEKIT_API_LEVEL "1")
 
+find_package(QT NAMES Qt6 Qt5 NO_MODULE REQUIRED COMPONENTS Core DBus)
+
 add_definitions(
     -DQT_NO_KEYWORDS
     -DQT_NO_CAST_TO_ASCII
@@ -53,7 +55,7 @@ set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 set (GETTEXT_PACKAGE "packagekit")
 set (VERSION "${QPACKAGEKIT_VERSION}")
 set (LOCALSTATEDIR "/var")
-set (CMAKECONFIG_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/packagekitqt5/")
+set (CMAKECONFIG_INSTALL_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/packagekitqt${QT_VERSION_MAJOR}/")
 
 add_definitions("-DLOCALSTATEDIR=\"${LOCALSTATEDIR}\"")
 set (CMAKE_CXX_STANDARD 11)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c559dc3..47b015b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 # CMakeLists for PackageKit-Qt library
 
-find_package(Qt5 5.6 REQUIRED COMPONENTS Core DBus)
+find_package(Qt${QT_VERSION_MAJOR} 5.10 REQUIRED COMPONENTS Core DBus)
 
 set(packagekitqt_HEADERS
     Daemon
@@ -41,25 +41,30 @@ endif ()
 set_source_files_properties(${PK_INTERFACE_XML} PROPERTIES NO_NAMESPACE true)
 set_source_files_properties(${PK_TRANSACTION_INTERFACE_XML} PROPERTIES NO_NAMESPACE true)
 
+if(${QT_VERSION_MAJOR} EQUAL 6)
+qt6_add_dbus_interface(packagekitqt_SRC ${PK_INTERFACE_XML} daemonproxy)
+qt6_add_dbus_interface(packagekitqt_SRC ${PK_TRANSACTION_INTERFACE_XML} transactionproxy)
+else()
 qt5_add_dbus_interface(packagekitqt_SRC ${PK_INTERFACE_XML} daemonproxy)
 qt5_add_dbus_interface(packagekitqt_SRC ${PK_TRANSACTION_INTERFACE_XML} transactionproxy)
+endif()
 
-add_library(packagekitqt5 ${packagekitqt_SRC} ${packagekitqt_HEADERS} ${packagekitqt_HEADERS_PRIVATE})
-set_target_properties(packagekitqt5 PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${QPACKAGEKIT_API_LEVEL})
+add_library(packagekitqt${QT_VERSION_MAJOR} ${packagekitqt_SRC} ${packagekitqt_HEADERS} ${packagekitqt_HEADERS_PRIVATE})
+set_target_properties(packagekitqt${QT_VERSION_MAJOR} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${QPACKAGEKIT_API_LEVEL})
 
-target_link_libraries(packagekitqt5 PUBLIC Qt5::DBus)
+target_link_libraries(packagekitqt${QT_VERSION_MAJOR} PUBLIC Qt${QT_VERSION_MAJOR}::DBus)
 
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/packagekitqt5.pc.in
-  ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt5.pc
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/packagekitqt${QT_VERSION_MAJOR}.pc.in
+  ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt${QT_VERSION_MAJOR}.pc
   @ONLY
 )
-target_include_directories(packagekitqt5 INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/packagekitqt5/PackageKit/;${CMAKE_INSTALL_INCLUDEDIR}/packagekitqt5>")
-install(TARGETS packagekitqt5 EXPORT PackageKitQtTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt5.pc
+target_include_directories(packagekitqt${QT_VERSION_MAJOR} INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/packagekitqt${QT_VERSION_MAJOR}/PackageKit/;${CMAKE_INSTALL_INCLUDEDIR}/packagekitqt${QT_VERSION_MAJOR}>")
+install(TARGETS packagekitqt${QT_VERSION_MAJOR} EXPORT PackageKitQtTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt${QT_VERSION_MAJOR}.pc
 	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
 )
 install(FILES ${packagekitqt_HEADERS}
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/packagekitqt5/PackageKit/
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/packagekitqt${QT_VERSION_MAJOR}/PackageKit/
 )
 
 add_subdirectory(modules)
diff --git a/src/modules/CMakeLists.txt b/src/modules/CMakeLists.txt
index 9ff1b46..74221d0 100644
--- a/src/modules/CMakeLists.txt
+++ b/src/modules/CMakeLists.txt
@@ -2,13 +2,13 @@ include(CMakePackageConfigHelpers)
 
 configure_package_config_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/packagekit-qt-config.cmake.in
-  ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt5-config.cmake
-  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/packagekitqt5/
+  ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt${QT_VERSION_MAJOR}-config.cmake
+  INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/packagekitqt${QT_VERSION_MAJOR}/
 )
 
-write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/packagekitqt5-config-version.cmake VERSION ${PROJECT_VERSION} COMPATIBILITY SameMajorVersion)
+write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/packagekitqt${QT_VERSION_MAJOR}-config-version.cmake VERSION ${PROJECT_VERSION} COMPATIBILITY SameMajorVersion)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt5-config.cmake
-        ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt5-config-version.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/packagekitqt5/
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt${QT_VERSION_MAJOR}-config.cmake
+        ${CMAKE_CURRENT_BINARY_DIR}/packagekitqt${QT_VERSION_MAJOR}-config-version.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/packagekitqt${QT_VERSION_MAJOR}/
 )
diff --git a/src/modules/packagekit-qt-config.cmake.in b/src/modules/packagekit-qt-config.cmake.in
index 6dc5f24..7a55fa2 100644
--- a/src/modules/packagekit-qt-config.cmake.in
+++ b/src/modules/packagekit-qt-config.cmake.in
@@ -1,7 +1,7 @@
 @PACKAGE_INIT@
 
-check_required_components(Qt5DBus)
+check_required_components(Qt@QT_VERSION_MAJOR@DBus)
 
-SET(PackageKitQt5_LIBRARIES "PK::packagekitqt5")
+SET(PackageKitQt@QT_VERSION_MAJOR@_LIBRARIES "PK::packagekitqt@QT_VERSION_MAJOR@")
 
 include("${CMAKE_CURRENT_LIST_DIR}/PackageKitQtTargets.cmake")
diff --git a/src/packagekitqt6.pc.in b/src/packagekitqt6.pc.in
new file mode 100644
index 0000000..4a34881
--- /dev/null
+++ b/src/packagekitqt6.pc.in
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+
+Name: @LIBNAME@
+Description: PackageKit is a package management abstraction layer.
+Version: @PROJECT_VERSION@
+Requires: Qt6Core, Qt6DBus
+Libs: -L${libdir} -lpackagekitqt6
+Cflags: -I${includedir}/packagekitqt6
-- 
2.36.1

